doc/developer/platform/formalism.md

```diff
@@ -55,8 +55,7 @@ SQL statements using a stateful *timeline* object. To map table names to
 `GlobalIds` and capabilities, Materialize maintains a *source* mapping. SQL
 indices are maintained by the compute layer.
 
-Finally: Materialize's adapter supports SQL transactions, but there are
-significant constraints on how these transactions can mix reads and writes.
```

Comment:
> but there are
significant constraints on how these transactions can mix reads and writes.

This goes against how this document is described in the beginning:
> This document details the **intended behavior** of the system.
It does not intend to describe the **current behavior** of the system.

I don't think that we believe that read/write transactions are fundamentally impossible in materialize. We don't currently support them because they are hard.